subsys/power/device.c

```diff
@@ -24,9 +24,8 @@ LOG_MODULE_DECLARE(power);
  */
 #if defined(CONFIG_SOC_FAMILY_NRF)
 #define MAX_PM_DEVICES	15
-#define NUM_CORE_DEVICES	4
 #define MAX_DEV_NAME_LEN	16
-static const char core_devices[NUM_CORE_DEVICES][MAX_DEV_NAME_LEN] = {
+static const char core_devices[][MAX_DEV_NAME_LEN] = {
```

Comment:
Sorry I didn't have time to review this.  While this isn't wrong, and is better than the previous approach, it still wastes memory compared to:
```
static const char * core_devices[] = { "CLOCK_32K", "CLOCK_16M", "sys_clock"};
```

Reply:
I agree, proposed version is even better :)